src/proxy/GmcpMessage.h

```diff
@@ -62,9 +66,14 @@ using GmcpJson = TaggedStringUtf8<tags::GmcpJsonTag>;
 class GmcpMessage final
 {
 private:
-    GmcpMessageName name;
-    std::optional<GmcpJson> json;
```

Comment:
Why give up on std::optional? Doing an empty check is basically the same thing but it's less obvious to the API user that its actually an optional payload according to the GMCP spec.